exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -1065,131 +1068,39 @@ func TestReconcileMachinePoolInfrastructure(t *testing.T) {
 }
 
 func TestReconcileMachinePoolMachines(t *testing.T) {
-	defaultCluster := clusterv1.Cluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      clusterName,
-			Namespace: metav1.NamespaceDefault,
-		},
-	}
+	t.Run("Reconcile MachinePool Machines", func(t *testing.T) {
```

Comment:
Is it possible to set this up with the test cases struct like before? I find that format is easier to understand and maintain when we need to add new tests. If not, I'm fine with this as is, it's non-blocking.

Reply:
It's not as straightforward but it could probably be done, I tried to follow other tests in the codebase that use envtest https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/machineset/machineset_controller_test.go#L75